sdk/src/lib.rs

```diff
@@ -115,6 +119,20 @@ where
         self
     }
 
+    // Set the max sink size.
+    pub fn with_max_message_size(mut self, message_size: String) -> Self {
+        self.max_message_size = Some(
+            message_size
+                .as_str()
+                .parse::<ByteSize>()
+                .unwrap()
+                .0
+                .try_into()
+                .unwrap(),
+        );
+        self
+    }
+
```

Comment:
Let's use size in bytes internally, and parse it once at the beginning.

```suggestion
    // Set the max message size.
    pub fn with_max_message_size(mut self, message_size: usize) -> Self {
        self.max_message_size = Some(message_size);
        self
    }
    
```